User login feature : Completed

ShashiSingh8434 opened the pull request:
@Deeptig9138 
All the features said in issue are now added in this branch.

You can just see the video and merge it without any issue.

To run the new project you will need a <serviceAccountKey.json> file from firebase for to merge the project with a valid realtime database : for that you can mail me personally to get that file so that you can run the project locally at your end or you can make one for yourself from : https://console.firebase.google.com/   

I didn't include <serviceAccountKey.json> in my commit for privacy reasons .

Just make a project and then create a web application realtime database in it then download the private unique .json file for yourself and place it at root folder of your project and change the <databaseURL> value with your new database link in database.py file. And you are good to go .

For further issue please do tell.

NOTE: There are still many things which other contributors can solve (like logout ,or making the UI better )as they are very easy 
           but time consuming. But this commit will provide with the feature without any problem.

Thank you !!!

Chechout the latest verion in working :
https://drive.google.com/drive/folders/1Qi13oX2N9SHV_rbZGfB3OEqx_eVryBqZ?usp=sharing

You can now merge this branch .
It's all done from my side.

ShashiSingh8434 commented:
@Deeptig9138 
Here is a video sample:
https://drive.google.com/drive/folders/1Qi13oX2N9SHV_rbZGfB3OEqx_eVryBqZ

ShashiSingh8434 commented:
@Deeptig9138 
And please add label and assignee to this PR also.
This PR can be merged without any issue and you can check the changed files and latest version of project in provided video samples.

Deeptig9138 commented:
First of all its great i saw the video. 

Now I understand this : I didn't include <serviceAccountKey.json> in my commit for privacy reasons, but as you have taken the work so i request that you create a normal one if possible without inputing the data as you have. Because this is not my expertise and thus it was never there in my project. 

Secondly your labels were already assigned before hand and you opened everything in that one issue and accordingly the label was given to you

and lastly on my end it shows that your branch has some conflicts to be resolved and then only i can merge the pull request. I am adding the screenshot for your knowledge. 
![Screenshot 2025-01-17 172820](https://github.com/user-attachments/assets/2d0b400f-8995-45a6-954d-502d1ede5ad8)

ShashiSingh8434 commented:
"> and lastly on my end it shows that your branch has some conflicts to be resolved and then only i can merge the pull request. I am adding the screenshot for your knowledge."

I understand that for you it all may look confusing since i added so many files and there is also some conflicts arised since you merged a PR from @KrishnashishMunshi .

I checked all files of main branch of the project after pulling it into my branch and then I modified my code for to resolve all issues so you can just accept all changes from my side it will not affect the project and also keep all changes that @KrishnashishMunshi made into the project .

If you face any more problem while merging please do tell .

ShashiSingh8434 commented:
And here is the link for <serviceAccountKey.json>
https://drive.google.com/file/d/1-Y2nd8nHwoQA-Viu9TWN-ZJqrbN9D4Er/view?usp=sharing

Download it and place it in root folder of project 
And make sure not to push it on GitHub (add the file name in .gitignore file if not already)

The file is in private mode so you will have to get access from me.

ShashiSingh8434 commented:
I have resolved the conflict. So you can merge now.


Here I have uploaded project directory folders tree for you to get all the file in correct order (download and paste the .json file for the project to run)

![Screenshot (52)](https://github.com/user-attachments/assets/d399df53-ca9a-4431-a7c1-4220a6b6b592)

![Screenshot 2025-01-17 201007](https://github.com/user-attachments/assets/c6d1ef16-707a-4ed9-b9cf-e7688315ca19)    ![Screenshot 2025-01-17 200941](https://github.com/user-attachments/assets/cba7dea3-c728-4c3d-b90a-bdc5370d0978)
